.github/workflows/checks.yml

```
@@ -21,7 +21,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
         with:
-          go-version: "1.22.11"
+          go-version: "1.22.12"
```

Instead of hard coding a Go version and bumping it with every patch release, would it possible to just take the Go version from `go.mod`?
```suggestion
         go-version-file: /path/to/go.mod
```

---

That would be easier to maintain, but this repository is a Go monorepo without a central Go modules file. Maybe referencing one of the internal packages would be fine (e.g `internal/tools/go.mod`). wdyt?

---

It's just something I have noticed while reading this change. I'm not aware of the Go version bump policy of this project. If the Go version in all go.mods is bumped at the same time, any one of the `go.mod`s could be referenced. 
So overall, I don't have a personal preference.

---

>  If the Go version in all go.mods is bumped at the same time, any one of the go.mods could be referenced.

At the moment, each submodule is bumped separately. I would prefer to wait until we have a `go.mod` file in the root directory that we can reference as base required Go version. Also, some CI checks run for two pinned Go versions to assert compatibility between the latest two minor versions.